lib/access_import/accessors/sprache_pro_hauptperson.rb

```diff
@@ -26,14 +26,10 @@ def level_keys
     ['Le', 'Sc', 'Sp', 'Ve'].map { |k| "fk_Kenntnisstufe#{k}".to_sym }
   end
 
-  def all
-    @records
-  end
-
   def where_person(person_id)
-    languages = @records.select do |key|
-      key.to_s == person_id.to_s
-    end
-    languages.map { |s| down_hkeys(s[1]) }
+    add_records_hash
+    @records.select do |_k, val|
+      val[:fk_Hauptperson].to_s == person_id.to_s
+    end.map { |_k, sk| down_hkeys(sk) }
```

Comment:
Please use more explicit variable names, I wouldn't know what `sk` is. Also `_key` is better than `_k`.